### PR TITLE
fixes #87: admin can now delete edit comments

### LINKED ISSUE #87: Discussion threads - user permissions
* Public users can only view/add/edit comments if the records are visible to the public
* For private/embargoed records, only users registered with that project can view/add/edit comments
* Only the comment author OR and administrator (ALA Admin/Project admin) can edit or delete a comment

### PR DESCRIPTION
**Pull request by temi**

### CODE REVIEW

File: grails-app/services/au/org/ala/biocollect/CommentService.groovy [hunk 1]
```diff
@@ -16,17 +16,25 @@ class CommentService {
     }
 
     def deleteComment(data){
-        webService.deleteWrapper(grailsApplication.config.ecodata.service.url + "/comment/${data['id']}?userId=${data['userId']}")
+        Boolean alaAdmin = authService.userInRole(grailsApplication.config.security.cas.adminRole)
```
Comment (chrisala) [inline grails-app/services/au/org/ala/biocollect/CommentService.groovy]: Are we supporting FC_ADMIN in BioCollect?  We have these kind of checks in UserService (e.g. isUserAlaOrFCAdmin)   Even if we aren't it's still nice if the UserService is the main place to make these checks / needs to know those config items)

File: web-app/js/comment.js [hunk 2]
```diff
@@ -351,8 +354,8 @@ function CommentListViewModel() {
     }
 
     // controls edit/delete button on a comment
-    self.isUserCommentOwner = function(comment){
-        return self.userId() == comment.userId()
+    self.canModifyDeleteComment = function(comment){
```
Comment (chrisala) [inline web-app/js/comment.js]: canModifyOrDeleteComment.

**Comment by chrisala** [comment]
As an opinion, do we want project admins to be able to edit user comments?  Do we flag on the UI or something that the comment has been edited by someone other than the creator?

File: web-app/js/comment.js (hunk 2 above)
Comment (temi) [inline web-app/js/comment.js]: done

File: grails-app/services/au/org/ala/biocollect/CommentService.groovy (hunk 1 above)
Comment (temi) [inline grails-app/services/au/org/ala/biocollect/CommentService.groovy]: I do not think we support FC_ADMIN. I have created a function userIsAlaAdmin in UserService.

**Comment by temi** [comment]
ideally, it would be nice to flag a comment by a moderator. But it would require some time to do it. I would prefer to have all the features we want implemented to be captured in an issue. Otherwise I feel this issue might never finish.

File: grails-app/controllers/au/org/ala/biocollect/merit/ProjectController.groovy [hunk 3]
```diff
@@ -169,169 +174,32 @@ class ProjectController {
...
+    def fields (){
+        params = [
+                'isSuitableForChildren', // child friendly
+                'difficulty', // difficulty level
+                'isDIY', // DIY
+                'status', 'endDate', // active check field status
+                'hasParticipantCost', // no cost
+                'hasTeachingMaterials', // teaching material
+                '', // mobile uses links to find it out
+                '', // page size
```
Comment (m-r-c) [inline grails-app/controllers/au/org/ala/biocollect/merit/ProjectController.groovy]: What do these empty strings do?

File: grails-app/services/au/org/ala/biocollect/merit/ProjectService.groovy [hunk 4]
```diff
@@ -15,7 +15,7 @@ class ProjectService {
     MetadataService metadataService
     SettingService settingService
 
-    def list(brief = false, citizenScienceOnly = false) {
+    def list(brief = false, citizenScienceOnly = false, filterP = null) {
```
Comment (m-r-c) [inline grails-app/services/au/org/ala/biocollect/merit/ProjectService.groovy]: Remove unused parameter ```filterP```

File: grails-app/controllers/au/org/ala/biocollect/merit/ProjectController.groovy (hunk 3 above)
Comment (temi) [inline grails-app/controllers/au/org/ala/biocollect/merit/ProjectController.groovy]: this should have been deleted. I missed it.

File: grails-app/services/au/org/ala/biocollect/merit/ProjectService.groovy (hunk 4 above)
Comment (temi) [inline grails-app/services/au/org/ala/biocollect/merit/ProjectService.groovy]: done

**Pull request closed without merging**